src/runtime/DateObject.cpp

```diff
@@ -1050,6 +1050,111 @@ int DateObject::yearFromTime(time64_t t)
     return estimate;
 }
 
+int DateObject::monthFromTime(time64_t t)
+{
+    int day = DateObject::daysInYear(yearFromTime(t));
+    bool leap = DateObject::inLeapYear(yearFromTime(t));
+
+    if (0 <= day && day < 31) {
+        return 0;
+    }
+
+    if (31 + leap <= day && day < 59 + leap) {
```

Comment:
If `day -= leap;` would be added, would it simplify the checks?

Isn't this check should be:
`if (31 <= day && day < 59 + leap)`

Another thing is the second part of the check is the same as the first part of the next check. The latter could be removed.

Something like this:

```
if (day < 31) {
  return 0;
}

day -= leap;

if (day < 59) {
  return 1;
}

if (day < 90) {
  return 2;
}
...
```

Reply:
Btw a logarithmic search could be added to speed up the code